### PR TITLE
refactor(scripts): reorganize build scripts

### PR DESCRIPTION
**Pull request by CITguy**
* `bin/<task>.js` -> `scripts/<task>.js`
* `lib/<module>` -> `scripts/_<module>/*`
* no change to logic other than corrections due to new file structure

### LGTM's
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit 3a3d79477322e1e07ed70a79b7aeba5b870d25fe

https://app.netlify.com/sites/helix-ui/deploys/5be20f289a063f402c09d5b9

File: .eslintignore [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
 bin/*
 lib/*
+scripts/*
 node_modules/*
 public/*
-source/vendor/*
 test/*
 
-# files
-_config.*
```
Comment (CITguy) [inline .eslintignore]: moved to `scripts/_config/*`

File: .eslintignore [hunk 2]
```diff
@@ -1,13 +1,10 @@
 # directories
 bin/*
 lib/*
+scripts/*
 node_modules/*
 public/*
-source/vendor/*
```
Comment (CITguy) [inline .eslintignore]: directory hasn't existed for a long time

File: .gitignore [hunk 3]
```diff
@@ -10,9 +10,7 @@ db.json
 public/
 .deploy*/
 dist/
-
-# Private credentials/config
-_config.private.js
+lib/
```
Comment (CITguy) [inline .gitignore]: reserved for built assets (`build` task will compile from `src` to `lib`)

File: .node-version [hunk 4]
```diff
@@ -1 +1 @@
-8.5.0
+10.12.0
```
Comment (CITguy) [inline .node-version]: As of 2018-10-30, Node 10.x is now in [Active Long-Term Support](https://github.com/nodejs/Release#release-schedule).

File: package.json [hunk 5]
```diff
@@ -65,27 +65,30 @@
...
-    "compile": "bin/compile.js",
+    "clean:public": "scripts/cleanPublic.js",
+    "compile": "scripts/compile.js",
     "test:coverage": "nyc --reporter=lcov mocha",
     "pregenerate": "yarn compile",
-    "generate": "bin/generate.js",
+    "generate": "scripts/generate.js",
     "preghpages": "yarn generate",
-    "ghpages": "bin/ghpages.js",
+    "ghpages": "scripts/publish.js",
```
Comment (CITguy) [inline package.json]: Haven't fully tested `ghpages` yet, but I'll address it when we perform the first release after this change is merged to `master`.

File: rollup.config.js [hunk 6]
```diff
@@ -4,11 +4,12 @@ import json from 'rollup-plugin-json';
 import pkg from './package.json';
```
Comment (cathysiller) [inline rollup.config.js]: move below

**Pull request merged**